cvmfs/catalog_mgr_client.cc

```diff
@@ -187,6 +187,9 @@ LoadError ClientCatalogManager::LoadCatalog(
     return success_code;
   }
 
+  if (manifest_.IsValid()) delete manifest_;
```

Comment:
That should not be necessary, it should be inherently done by the unique pointer